packages/flutter_tools/lib/src/host_validator.dart

```diff
@@ -0,0 +1,93 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:http/http.dart' as http;
+import 'package:meta/meta.dart';
+import 'base/platform.dart';
+import 'doctor.dart';
+
+/// Common Flutter hosts.
+const List<String> commonRequiredHosts = <String>[
+  'https://pub.dev/',
+  'https://cloud.google.com/',
+  'https://maven.google.com/',
+];
+
+/// MacOS specific required hosts.
+const List<String> macOSRequiredHosts = <String>[
+  'https://cocoapods.org/',
+];
+
+/// Verify that the required hosts stored at [commonRequiredHosts] and [macOSRequiredHosts] are reachable.
+class HostValidator extends DoctorValidator {
+  HostValidator({
+    @required Platform platform,
+    http.Client httpClient,
+  })  : _platform = platform,
+        _httpClient = httpClient ?? http.Client(),
```

Comment:
Can you make this parameter required and remove the fallback? We have had issues in the past with tests not providing certain dependencies and then quietly the fallback is used.

In the places in the codebase where there are fallbacks, it is usually to make an API change non-breaking.